samples/metricsBot/metricsMiddleware.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/PaulSonOfLars/gotgbot/v2"
+)
+
+var (
+	totalRequests = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "gotgbot",
+			Name:      "requests_total",
+			Help:      "Number of requests made to the bot API.",
+		},
+		[]string{
+			"api_method",
+		},
+	)
+	totalHTTPErrors = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "gotgbot",
+			Name:      "http_request_errors_total",
+			Help:      "Number of HTTP errors obtained.",
+		},
+		[]string{
+			"api_method",
+		},
+	)
+	totalAPIErrors = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "gotgbot",
+			Name:      "api_request_errors_total",
+			Help:      "Number of bot API errors obtained.",
+		},
+		[]string{
+			"api_method",
+			"api_status_code",
+			"description",
+		},
+	)
+	requestDuration = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "gotgbot",
+			Name:      "api_request_time_seconds",
+			Help:      "Duration of requests made to the bot API.",
+		},
+		[]string{
+			"api_method",
```

Comment:
http_method label is missing in this metric

Reply:
I removed the http_method label because the bot client only uses POST methods; so the field ends up being redundant. 

Additionally, that's not information that gets exposed in the error returned, so I figured it would be best to remove it rather than hardcode it